keras/models.py

```diff
@@ -233,61 +233,60 @@ def convert_custom_objects(obj):
         if obj in custom_objects:
             return custom_objects[obj]
         return obj
+    with h5py.File(filepath, mode='r') as f:
+        # instantiate model
+        model_config = f.attrs.get('model_config')
+        if model_config is None:
+            raise ValueError('No model found in config file.')
+        model_config = json.loads(model_config.decode('utf-8'))
+        model = model_from_config(model_config, custom_objects=custom_objects)
+
+        # set weights
+        topology.load_weights_from_hdf5_group(f['model_weights'], model.layers)
+
+        # Early return if compilation is not required.
+        if not compile:
+            f.close()
```

Comment:
Returning would close the file (I think) since you're already in a 'with' statement.